internal/servers/controller/handlers/authmethods/authmethod_service.go

```diff
@@ -246,12 +302,32 @@ func (s Service) UpdateAuthMethod(ctx context.Context, req *pbs.UpdateAuthMethod
 			return nil, err
 		}
 	}
-	u.Scope = authResults.Scope
-	u.AuthorizedActions = authResults.FetchActionSetForId(ctx, u.Id, IdActions[auth.SubtypeFromId(u.Id)]).Strings()
-	if err := populateCollectionAuthorizedActions(ctx, authResults, u); err != nil {
+
+	outputFields := requests.OutputFields(ctx)
+	item, err := toAuthMethodProto(ctx, am, handlers.WithOutputFields(&outputFields))
+	if err != nil {
 		return nil, err
 	}
-	return &pbs.UpdateAuthMethodResponse{Item: u}, nil
+
+	if outputFields.Has(globals.ScopeField) {
+		item.Scope = authResults.Scope
+	}
+	if outputFields.Has(globals.AuthorizedActionsField) {
+		item.AuthorizedActions = authResults.FetchActionSetForId(ctx, am.GetPublicId(), IdActions[auth.SubtypeFromId(am.GetPublicId())]).Strings()
+	}
+	if outputFields.Has(globals.AuthorizedCollectionActionsField) {
+		collectionActions, err := populateCollectionAuthorizedActions(ctx, authResults, am)
+		if err != nil {
+			return nil, err
+		}
+		item.AuthorizedCollectionActions = collectionActions
+	}
+
+	if item.GetAttributes() != nil && dryRun {
```

Comment:
The more I look at this the more I think it would be better if it could be handled in the toProto method.  That is the whole purpose of that, and now that it accepts a variadic option param it probably isn't hard to add these and the authorized action stuff.

Reply:
I'll address this in the next PR.